pymc_marketing/model_config.py

```diff
@@ -0,0 +1,484 @@
+        "kwargs": {
+            "mu": 0,
+            "sigma": 1,
+        },
+        # dims need to be specified now!
+        "dims": "channel",
+    }
+
+Example configuration of 1D parameter with a hierarchical distribution for
+only the mu:
+
+.. code-block:: python
+
+    hierarchical_parameter = {
+        "dist": "Normal",
+        "kwargs": {
+            # Replace the scalar parameter values with additional distributions
+            "mu": {
+                "dist": "Normal",
+                "kwargs": {
+                    "mu": 0,
+                    "sigma": 1,
+                },
+            },
+            # Common sigma for all channels
+            "sigma": 1,
+        },
+        "dims": "channel",
+    }
+
+Example configuration of a 2D parameter:
+
+.. code-block:: python
+
+    matrix_parameter = {
+        "dist": "Normal",
+        "kwargs": {
+            "mu": 0,
+            "sigma": 1,
+        },
+        # dims need to be specified now!
+        "dims": ("channel", "geo"),
+    }
+
+Model configuration with all of these variables:
+
+.. code-block:: python
+
+    model_config = {
+        "alpha": scalar_parameter,
+        "beta": vector_parameter,
+        "gamma": hierarchical_parameter,
+        "delta": matrix_parameter,
+    }
+
+Creating variables from the configuration:
+
+.. code-block:: python
+
+    import pymc as pm
+
+    from pymc_marketing.model_config import create_distribution_from_config
+
+    coords = {
+        "channel": ["A", "B", "C"],
+        "geo": ["Region1", "Region2"],
+    }
+    with pm.Model(coords=coords) as model:
+        alpha = create_distribution_from_config("alpha", model_config)
+        beta = create_distribution_from_config("beta", model_config)
+        gamma = create_distribution_from_config("gamma", model_config)
+        delta = create_distribution_from_config("delta", model_config)
+
+"""
+
+from collections.abc import Callable
+from copy import deepcopy
+from typing import Any
+
+import numpy as np
+import pandas as pd
+import pymc as pm
+from pymc.distributions.shape_utils import Dims
+from pytensor import tensor as pt
+
+
+class UnsupportedShapeError(Exception):
+    """Error for when the shape of the hierarchical variable is not supported."""
+
+
+def handle_scalar(var: pt.TensorLike, dims: Dims, desired_dims: Dims) -> pt.TensorLike:
+    """Broadcast a scalar to the desired dims."""
+    return var
+
+
+def handle_1d(var: pt.TensorLike, dims: Dims, desired_dims: Dims) -> pt.TensorLike:
+    """Broadcast a 1D variable to the desired dims."""
+    return var
+
+
+def handle_2d(var: pt.TensorLike, dims: Dims, desired_dims: Dims) -> pt.TensorLike:
+    """Broadcast a 2D variable to the desired dims."""
+    if dims == desired_dims:
+        return var
+
+    if dims[::-1] == desired_dims:
+        return var.T
+
+    if dims[0] == desired_dims[-2]:
+        return var[:, None]
+
+    return var
+
+
+HANDLE_MAPPING = {0: handle_scalar, 1: handle_1d, 2: handle_2d}
```

Comment:
Could an `Enum` be a better structure?

Reply:
(so that is not mutable?)

Reply:
I used this: https://stackoverflow.com/questions/11014262/how-to-create-an-immutable-dictionary-in-python
It will stay the same after initialization but act like a dictionary still